red/runtime/nodes/context/localfilesystem.js

```diff
@@ -213,10 +228,16 @@ LocalFileSystem.prototype.open = function(){
 }
 
 LocalFileSystem.prototype.close = function(){
-    if (this.cache && this._flushPendingWrites) {
+    var self = this;
+    if (this.cache && this._pendingWriteTimeout) {
         clearTimeout(this._pendingWriteTimeout);
         delete this._pendingWriteTimeout;
-        return this._flushPendingWrites();
+        this.flushInterval = 0;
+        return this.writePromise.then(function(){
+            if(Object.keys(self.pendingWrites).length > 0) {
```

Comment:
As stated in my previous comment - I'm not sure this block will be needed if we move where `_pendingWriteTimeout` is deleted.